test/pom.xml

```diff
@@ -226,6 +226,25 @@
             </systemPropertyVariables>
           </configuration>
         </plugin>
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-jar-plugin</artifactId>
+          <executions>
+            <execution>
+              <goals>
+                <goal>test-jar</goal>
+              </goals>
+              <phase>pre-integration-test</phase>
```

Comment:
Might be better to just bind this to test-compile to let `mvn test -Dtest=ScannerContextIT` still work. Creating the jar won't be too much of an impact, I guess.